src/puppetlabs/comidi.clj

```diff
@@ -21,17 +21,24 @@
 (def Zipper
   (schema/pred ks/zipper?))
 
-(def http-methods
-  #{:any :get :post :put :delete :head :options})
-
 (def RequestMethod
   (schema/enum :any :get :post :put :delete :head :options))
 
+; Derived from bidi-schema PatternSegment
 (def RegexPatternSegment
-  [(schema/one Pattern "regex") (schema/one schema/Keyword "variable")])
+  (schema/pair schema/Regex "qual" schema/Keyword "id"))
+
+; Derived from bidi-schema by flattening a vector of Patterns
+(def RouteInfoPathSegment
+  (schema/either bidi-schema/Path
+                 bidi-schema/PatternSegment
+                 schema/Bool))
```

Comment:
The picture tries to compare the error messages. As far as I can see we're getting horrible errors either way - the left pane of the screenshot is ```either``` and the right pane is ```cond```. I'm not sure there's an improvement - unless there's a better way to write the ```cond``` than the one I took.